test/parallel/test-readline-async-iterators-destroy.js

```diff
@@ -64,9 +64,13 @@ async function testMutualDestroy() {
     const iteratedLines = [];
     for await (const k of rli) {
       iteratedLines.push(k);
-      for await (const l of rli) {
-        iteratedLines.push(l);
-        break;
+      try {
+        for await (const l of rli) {
+          iteratedLines.push(l);
+          break;
+        }
+      } catch (err) {
+        assert.strictEqual(err.code, 'ERR_STREAM_ITERATOR_EXISTS');
```

Comment:
I think this function can be simplified to just test for `ERR_STREAM_ITERATOR_EXISTS`, the rest of the logic is already covered in the previous functions.